aws/cloudformation/ami-manager.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file AWS Lambda Custom Resource function to manage automatic AMI creation.
+ * @see {@link https://blogs.aws.amazon.com/application-management/post/Tx38Z5CAM5WWRXW|Faster Auto Scaling in AWS CloudFormation Stacks with Lambda-backed Custom Resources}
+ */
+
+var response = require('cfn-response');
+
+/** Takes an AWS CloudFormation stack name and instance id and returns the newly-created AMI ID. **/
+exports.handler = function (event, context) {
+  console.log("REQUEST RECEIVED:\n", JSON.stringify(event));
+
+  var stackName = event.ResourceProperties.StackName;
+  var instanceId = event.ResourceProperties.InstanceId;
+  var instanceRegion = event.ResourceProperties.Region;
+
+  var responseStatus = "FAILED";
+  var responseData = {};
+  var physicalId = event.PhysicalResourceId;
+
+  function error(err, msg) {
+    responseData = {Error: msg};
+    console.log(responseData.Error + ":\n", err);
+    response.send(event, context, responseStatus, responseData, physicalId);
+  }
+
+  var AWS = require("aws-sdk");
+  var ec2 = new AWS.EC2({region: instanceRegion});
+
+  console.log("REQUEST TYPE:", event.RequestType);
+  if (event.RequestType == "Delete") {
+    var params = {
+      ImageIds: [ event.PhysicalResourceId ]
+    };
+    ec2.describeImages(params, function (err, data) {
+      if (err) {
+        error(err, "DescribeImages call failed");
+      } else if (data.Images.length === 0) {
+        response.send(event, context, "SUCCESS", {Info: "Nothing to delete"});
+      } else {
+        var imageId = data.Images[0].ImageId;
```

Comment:
This seems to assume that length is exactly 1.  Is it correct to ignore the other images?